scripts/quests/sandoria/Her_Majesty_S_Garden.lua

```diff
@@ -0,0 +1,102 @@
+-----------------------------------
+-- Her Majestys Garden
+-----------------------------------
+-- Log ID: 0, Quest ID: 62
+-- Chalvatot : !gotoid 17731598
+-----------------------------------
+require('scripts/globals/npc_util')
+require('scripts/globals/items')
+require('scripts/globals/keyitems')
+require('scripts/globals/quests')
+require('scripts/globals/status')
+require('scripts/globals/zone')
+require('scripts/globals/interaction/quest')
+-----------------------------------
+local ID = require("scripts/zones/Chateau_dOraguille/IDs")
+
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.SANDORIA, xi.quest.id.sandoria.HER_MAJESTY_S_GARDEN)
+
+quest.reward =
+{
+    fame     = 30,
+    fameArea = xi.quest.fame_area.SANDORIA,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+            player:getFameLevel(xi.quest.fame_area.SANDORIA) >= 4
+        end,
+
+        [xi.zone.CHATEAU_DORAGUILLE] =
+        {
+            ['Chalvatot'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 0 and player:getNation() ~= xi.nation.SANDORIA then -- Non sandy folk gets a CS prior to getting acceptance
+                        return quest:progressEvent(583)
+                    elseif questProgress == 0 and player:getNation() == xi.nation.SANDORIA then -- Sandy folks go directly to CS acceptance
+                        return quest:progressEvent(84)
+                    elseif questProgress == 1 then
+                        return quest:progressEvent(84)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [583] = function(player, csid, option, npc)
+                    return quest:setVar(player, 'Prog', 1)
+                end,
+
+                [84] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:begin(player)
+                        return quest:setVar(player, 'Prog', 0)
```

Comment:
Shouldn't be returning the setting of variables onEventFinish. 
Only need to return quest object on triggers or trades, etc. onEventFinish is the handling of those events